src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -436,12 +436,26 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
 
                 // because we are using dns name instead of service we have to retrieve it by splitting when needed
                 // If this ever cause issues we should consider larger refactor where we add serviceName member variable to EndpointInfo class.
-                var serviceName = endpoint.DnsName
+                var dnsNameArray = endpoint.DnsName
                     .ToUpperInvariant()
-                    .Split(".")
+                    .Split(".");
+
+                var serviceName = dnsNameArray
                     .First()
-                    .Replace("-", "_")
-                    .Replace(".", "_");

```

Comment:
this was redundant regardless of my changes

Reply:
@garyburgmann I agree is there any changes when dns name is  abc.companyname.com ? in that case you will need Replace(".", "_") ? Just thinking out aloud. Thank you for this PR.

Reply:
@hsubramanianaks `.First()` has been around since before my changes. So in your example the string is simply `abc` that we are acting on.